bullet/src/EntityManagementFeatures.cc

```diff
@@ -33,31 +33,28 @@ Identity EntityManagementFeatures::ConstructEmptyWorld(
 bool EntityManagementFeatures::RemoveModel(const Identity &_modelID)
 {
   // Check if the model exists
-  if (this->models.find(_modelID) == this->models.end()){
+  if (this->models.find(_modelID.id) == this->models.end()){
     return false;
   }
-
-  const auto &modelInfo = this->models.at(_modelID);
-
   // Current implementation does not include collisions nor joints
   // Those should be removed here before removing the model
 
   // Clean up links
-  for (const auto &linkEntry : this->links)
+  std::unordered_map<std::size_t, LinkInfoPtr>::iterator it = this->links.begin();
+  while (it != this->links.end())
   {
-    const auto &linkInfo = linkEntry.second;
+    const auto &linkInfo = it->second;
     if (linkInfo->model.id == _modelID.id)
     {
-      this->links.erase(linkEntry.first);
+      it = this->links.erase(it);
+      continue;
     }
+    it++;
   }
 
-  // (To-DO blast545): test if its required to make an extra call to
-  // world->removeMultiBody(model) to remove it from the bullet system
-  // or is it enough just by deleting its pointed memory
-
   // Clean up model
-  delete modelInfo->model;
+  this->worlds.at(this->models.at(_modelID)->world)->world->removeMultiBody(this->models.at(_modelID)->model);
```

Comment:
In line 36, the get to the model is `this->models.find(_modelID.id)`, it seems to me the call here should be replaced to match that behavior as well (using id instead of identity)
```suggestion
  this->worlds.at(this->models.at(_modelID.id)->world).id->world->removeMultiBody(this->models.at(_modelID.id)->model);
```

Reply:
Although the key access of the unordered map, are `size_t` and internally the compiler is making the conversion. Maybe just remove the `id` in line 36 to have them using Identity

Reply:
(disregard this comment in favor to the suggestion for line 36)

Reply:
Updated on 205fcc8, great catch